Traceback (most recent call last):
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
dsets = []
for device in devices:
    try:
        dsets.append(process_station(device))
    except:
        pass
    time.sleep(5)
ds = xr.concat(dsets, dim='station')
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mStopIteration[0m                             Traceback (most recent call last)
File [0;32m~/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/xarray/structure/concat.py:254[0m, in [0;36mconcat[0;34m(objs, dim, data_vars, coords, compat, positions, fill_value, join, combine_attrs, create_index_for_new_dim)[0m
[1;32m    253[0m [38;5;28;01mtry[39;00m:
[0;32m--> 254[0m     first_obj, objs [38;5;241m=[39m [43mutils[49m[38;5;241;43m.[39;49m[43mpeek_at[49m[43m([49m[43mobjs[49m[43m)[49m
[1;32m    255[0m [38;5;28;01mexcept[39;00m [38;5;167;01mStopIteration[39;00m [38;5;28;01mas[39;00m err:

File [0;32m~/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/xarray/core/utils.py:241[0m, in [0;36mpeek_at[0;34m(iterable)[0m
[1;32m    240[0m gen [38;5;241m=[39m [38;5;28miter[39m(iterable)
[0;32m--> 241[0m peek [38;5;241m=[39m [38;5;28;43mnext[39;49m[43m([49m[43mgen[49m[43m)[49m
[1;32m    242[0m [38;5;28;01mreturn[39;00m peek, itertools[38;5;241m.[39mchain([peek], gen)

[0;31mStopIteration[0m: 

The above exception was the direct cause of the following exception:

[0;31mValueError[0m                                Traceback (most recent call last)
Cell [0;32mIn[6], line 8[0m
[1;32m      6[0m         [38;5;28;01mpass[39;00m
[1;32m      7[0m     time[38;5;241m.[39msleep([38;5;241m5[39m)
[0;32m----> 8[0m ds [38;5;241m=[39m [43mxr[49m[38;5;241;43m.[39;49m[43mconcat[49m[43m([49m[43mdsets[49m[43m,[49m[43m [49m[43mdim[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43mstation[39;49m[38;5;124;43m'[39;49m[43m)[49m

File [0;32m~/miniconda3/envs/instrument-cookbooks-dev/lib/python3.10/site-packages/xarray/structure/concat.py:256[0m, in [0;36mconcat[0;34m(objs, dim, data_vars, coords, compat, positions, fill_value, join, combine_attrs, create_index_for_new_dim)[0m
[1;32m    254[0m     first_obj, objs [38;5;241m=[39m utils[38;5;241m.[39mpeek_at(objs)
[1;32m    255[0m [38;5;28;01mexcept[39;00m [38;5;167;01mStopIteration[39;00m [38;5;28;01mas[39;00m err:
[0;32m--> 256[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m([38;5;124m"[39m[38;5;124mmust supply at least one object to concatenate[39m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m[38;5;250m [39m[38;5;21;01merr[39;00m
[1;32m    258[0m [38;5;28;01mif[39;00m compat [38;5;129;01mnot[39;00m [38;5;129;01min[39;00m [38;5;28mset[39m(_VALID_COMPAT) [38;5;241m-[39m {[38;5;124m"[39m[38;5;124mminimal[39m[38;5;124m"[39m}:
[1;32m    259[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m    260[0m         [38;5;124mf[39m[38;5;124m"[39m[38;5;124mcompat=[39m[38;5;132;01m{[39;00mcompat[38;5;132;01m!r}[39;00m[38;5;124m invalid: must be [39m[38;5;124m'[39m[38;5;124mbroadcast_equals[39m[38;5;124m'[39m[38;5;124m, [39m[38;5;124m'[39m[38;5;124mequals[39m[38;5;124m'[39m[38;5;124m, [39m[38;5;124m'[39m[38;5;124midentical[39m[38;5;124m'[39m[38;5;124m, [39m[38;5;124m'[39m[38;5;124mno_conflicts[39m[38;5;124m'[39m[38;5;124m or [39m[38;5;124m'[39m[38;5;124moverride[39m[38;5;124m'[39m[38;5;124m"[39m
[1;32m    261[0m     )

[0;31mValueError[0m: must supply at least one object to concatenate

